conan_provider.cmake

```diff
@@ -477,13 +477,18 @@ macro(conan_provide_dependency method package_name)
         construct_profile_argument(_host_profile_flags CONAN_HOST_PROFILE)
         construct_profile_argument(_build_profile_flags CONAN_BUILD_PROFILE)
         get_property(_MULTICONFIG_GENERATOR GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+        if(EXISTS "${CMAKE_SOURCE_DIR}/conanfile.py")
+            set(generator "")
+        else ()
+            set(generator "-g;CMakeDeps")
```

Comment:
For the ``conanfile.txt`` I'd try:
- Search for the "CMakeDeps" string in the conanfile.txt
- If the string is not present, then display a warning, saying that it should be there, and this will be changed in the future, and add the generator.

I think that having the generator in the ``conanfile.txt`` can also help to switch to the ``conan install`` + ``cmake`` that might be more typical in CI, for example, so I'd aim for that for the future.

Reply:
Okey, so the idea is that the CMakeDeps should be declared in the conanfile.txt and we want it to be the standard in the future.
Maybe it would be interesting to have the warning  for the conanfile.py too since there is no warning or specific error if there's no CMakeDeps generator declared. It will just fail when running cmake.

Reply:
There is a valid use case that is users that don't want to use the CMakeDeps generated files but they are using the packaged xxx-config.cmake inside the packages. It would be a valid use case (we might need to add more functionality to the provider and collect the dependencies build-folders to pass them to CMAKE_PREFIX_PATH) that we might want to support in the future. So aiming to making it always a user explicitly defined things seems more flexible and less problematic.